### PR TITLE
polkadot: 1.0.0 -> 1.1.0

### PR DESCRIPTION
**Pull request by andresilva**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by andresilva** [comment]
We can now build the runtimes since rustc wasm32-unknown-unknown target is now available in nixpkgs.

File: pkgs/applications/blockchains/polkadot/default.nix [hunk 1]
```diff
@@ -31,48 +32,41 @@ rustPlatform.buildRustPackage rec {
     '';
   };
 
+  preBuild = ''
+    export SUBSTRATE_CLI_GIT_COMMIT_HASH=$(cat .git_commit)
```
Comment (asymmetric) [inline pkgs/applications/blockchains/polkadot/default.nix:36]: Bash has built-in support for this: https://www.gnu.org/software/bash/manual/html_node/Command-Substitution.html

```suggestion
    export SUBSTRATE_CLI_GIT_COMMIT_HASH=$(< .git_commit)
```

File: pkgs/applications/blockchains/polkadot/default.nix [hunk 2]
```diff
@@ -31,48 +32,41 @@ rustPlatform.buildRustPackage rec {
...
-    FAST_UNSTAKE_DOCIFY_DIR=$FAST_UNSTAKE_DIR/frame/fast-unstake
+  cargoBuildFlags = [ "-p" "polkadot" ];
 
-    mkdir -p $FAST_UNSTAKE_DOCIFY_DIR
-    cp -r $FAST_UNSTAKE_DIR/src $FAST_UNSTAKE_DOCIFY_DIR
-  '';
-
-  buildInputs = lib.optionals stdenv.isDarwin [ Security SystemConfiguration ];
+  # NOTE: tests currently fail to compile due to an issue with cargo-auditable
+  # and resolution of features flags, I think related to this:
```
Comment (asymmetric) [inline pkgs/applications/blockchains/polkadot/default.nix:53]:
```suggestion
  # and resolution of features flags, potentially related to this:
```

File: pkgs/applications/blockchains/polkadot/default.nix [hunk 3]
```diff
@@ -31,48 +32,41 @@ rustPlatform.buildRustPackage rec {
...
+  # https://github.com/rust-secure-code/cargo-auditable/issues/66
+  doCheck = false;
 
-  nativeBuildInputs = [ rustPlatform.bindgenHook ];
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    rustc-wasm32
+    rustc-wasm32.llvmPackages.lld
```
[the remark this answers is not in the document]
Comment (andresilva) [reply pkgs/applications/blockchains/polkadot/default.nix:60]: The target platform for the runtime is always Wasm, the polkadot binary includes a Wasm VM. If you want to learn more about this check here: https://docs.substrate.io/learn/architecture/

**Review by asymmetric — APPROVED** [review]

**Pull request merged**